project2-beat-that/script.js

```diff
@@ -1,4 +1,266 @@
+// 2 players; 2nd goes after the 1st
+// Each player will roll 2 dices.
+// [1] To start the game, Player 1 will click submit button to roll their dices.
+// [2] Upon clicking submit button, Player 1 will be shown (output) their dice roll numbers.
+// [3] Player 1 will then input 1 or 2 to determine their score.
+// [3.1] how score will be tabulated: if input is 1, dice 1 and dice 2 will be concatenated; if input is 2, dice 2 and dice 1 will be concatenated.
+// [4] Repeat steps 1 to 3 for player 2.
+// [5] Program will finally compare the scores and show result.
+// [5.1] list out all possible scenarios (win, lose, draw).
+
+var gameMode = "player1diceroll";
+
+//initialize array to store 2 dices rolled.
+var dicesRolled = [];
+// var message = " ";
+
+//initialize first player.
+var currentPlayer = 1;
+
+// Player 1 and Player 2 score.
+var player1Score = 0;
+var player2Score = 0;
+
+//randomly generate a number  from 1 to 6
+var generateRandomNumber = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//push the dice number of each of 2 rolls into the array
+var get2DiceRolls = function () {
+  dicesRolled = [];
+  for (var i = 0; i < 2; i += 1) {
+    dicesRolled.push(generateRandomNumber());
+  }
+  return dicesRolled;
+};
+
+var getPlayer1Score = function (userInput) {
+  let firstDice = dicesRolled[0];
+  let secondDice = dicesRolled[1];
+  if (userInput === 1) {
+    player1Score = Number(String(firstDice) + String(secondDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+  if (userInput === 2) {
+    player1Score = Number(String(secondDice) + String(firstDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
```

Comment:
good but notice that the return statements are the same, perhaps we can just have an overall return at the end?

Reply:
also since you have input validation in place, then userInput is bound to be 1 or 2. so theres no need to check explicitly for `userInput === 2`, it can just be an `else`